### PR TITLE
ISSUE #4234 - Fix oneOf and manyOf crashing app for values of riskCategories and jobsAndUsers

### LINKED ISSUE #4234: Viewer white screens when I try to create a ticket with a specific template
### Description
Try to create a ticket with the following template results in a whitescreen: 

````json
{
  "_id": "eedc69e0-fa55-11ed-b770-bf11c048ba35",
  "modules": [
    {
      "type": "sequencing",
      "properties": []
    },
    {
      "type": "safetibase",
      "properties": []
    },
    {
      "properties": [
        {
          "name": "text",
          "type": "text"
        },
        {
          "name": "long text",
          "type": "longText"
        },
        {
          "type": "boolean",
          "default": false,
          "name": "boolean"
        },
        {
          "name": "date",
          "type": "date"
        },
        {
          "name": "number",
          "type": "number"
        },
        {
          "name": "oneOf",
          "type": "oneOf",
          "values": [
            "Apple",
            "Orange",
            "Banana"
          ]
        },
        {
          "name": "oneOf - jobs and users",
          "type": "oneOf",
          "values": "jobsAndUsers"
        },
        {
          "name": "oneOf - Risk Cat",
          "type": "oneOf",
          "values": "riskCategories"
        },
        {
          "name": "manyOf",
          "type": "manyOf",
          "values": [
            "Apple",
            "Orange",
            "Banana"
          ]
        },
        {
          "name": "many - jobs and users",
          "type": "manyOf",
          "values": "jobsAndUsers"
        },
        {
          "name": "manyOf - Risk Cat",
          "type": "manyOf",
          "values": "riskCategories"
        },
        {
          "name": "image",
          "type": "image"
        },
        {
          "name": "view",
          "type": "view"
        }
      ],
      "name": "Custom module"
    }
  ],
  "properties": [
    {
      "name": "text",
      "type": "text"
    },
    {
      "name": "long text",
      "type": "longText"
    },
    {
      "type": "boolean",
      "default": false,
      "name": "boolean"
    },
    {
      "name": "date",
      "type": "date"
    },
    {
      "name": "number",
      "type": "number"
    },
    {
      "name": "oneOf",
      "type": "oneOf",
      "values": [
        "Apple",
        "Orange",
        "Banana"
      ]
    },
    {
      "name": "oneOf - jobs and users",
      "type": "oneOf",
      "values": "jobsAndUsers"
    },
    {
      "name": "oneOf - Risk Cat",
      "type": "oneOf",
      "values": "riskCategories"
    },
    {
      "name": "manyOf",
      "type": "manyOf",
      "values": [
        "Apple",
        "Orange",
        "Banana"
      ]
    },
    {
      "name": "many - jobs and users",
      "type": "manyOf",
      "values": "jobsAndUsers"
    },
    {
      "name": "manyOf - Risk Cat",
      "type": "manyOf",
      "values": "riskCategories"
    },
    {
      "name": "image",
      "type": "image"
    },
    {
      "name": "view",
      "type": "view"
    }
  ],
  "config": {
    "pin": true,
    "defaultView": true,
    "issueProperties": true,
    "comments": true
  },
  "code": "ALL",
  "name": "All Fields"
}
````

### Steps to replicate
- add this template to a teamspace
- Open a model, try to create a new ticket with this template

### Current Behaviour
- viewer white screen as soon as you select the template after `New Ticket`
image

### Expected Behaviour
- should allow me to create a ticket as it is (should be?) a valid template

### Pictures/ Gifs of the error

img

### PR DESCRIPTION
**Pull request by The-Daniel**
This fixes #4234 

#### Description
- Stopped the app crashing when custom properties of a oneOf or manyOf type have the value riskCategories or jobsAndUsers
- jobsAndUsers custom properties now get the new assignees selects
- Fixed a bug where multi selects weren't showing their value

#### Test cases
- Use a template with these custom values (an example template is offered below)
- Change the values of all the different selects

<details><summary>Ticket template with all selects</summary>
<p>

```{
  "modules": [
    {
      "name": "Required selects",
      "deprecated": false,
      "properties": [
        {
      "type": "oneOf",
      "values": "jobsAndUsers",
      "name": "User/Job OneOf",
"required": true
    },
    {
      "type": "manyOf",
      "values": "jobsAndUsers",
      "name": "User/Job ManyOf",
"required": true
    },
    {
      "type": "oneOf",
      "values": "riskCategories",
      "name": "Risky OneOf",
"required": true
    },
    {
      "type": "manyOf",
      "values": "riskCategories",
      "name": "Risky ManyOf",
"required": true
    },
    {
      "type": "oneOf",
      "values": [
        "Space is the Place",
        "Discipline 33",
        "Rocket Number 9"
      ],
      "name": "Normal OneOf",
"required": true
    },
    {
      "type": "manyOf",
      "values": [
        "Space is the Place",
        "Discipline 33",
        "Rocket Number 9"
      ],
      "name": "Normal ManyOf",
"required": true
    }
       ]
    }
],
  "properties": [
    {
      "type": "oneOf",
      "values": "jobsAndUsers",
      "name": "User/Job OneOf"
    },
    {
      "type": "manyOf",
      "values": "jobsAndUsers",
      "name": "User/Job ManyOf"
    },
    {
      "type": "oneOf",
      "values": "riskCategories",
      "name": "Risky OneOf"
    },
    {
      "type": "manyOf",
      "values": "riskCategories",
      "name": "Risky ManyOf"
    },
    {
      "type": "oneOf",
      "values": [
        "Space is the Place",
        "Discipline 33",
        "Rocket Number 9"
      ],
      "name": "Normal OneOf"
    },
    {
      "type": "manyOf",
      "values": [
        "Space is the Place",
        "Discipline 33",
        "Rocket Number 9"
      ],
      "name": "Normal ManyOf"
    }
  ],
  "config": {
    "issueProperties": true
  },
  "code": "SEL",
  "name": "Selects"
}```

</p>
</details>

### CODE REVIEW

**Comment by Amantini1997** [comment]
This is a single select, but all the options have the checkbox, is this correct? 
image

File: frontend/src/v5/ui/controls/assigneesSelect/assigneesSingleSelect.component.tsx
Comment (Amantini1997) [inline frontend/src/v5/ui/controls/assigneesSelect/assigneesSingleSelect.component.tsx:1]: Most of the code in this file is the same as "assigneesMultiSelect", the major differences I have seen are:
- the `value` (single vs multiple)
- the `multiple` flag
- some content
Screenshot 2023-05-30 171145


Maybe some of the shared can be moved inside `assigneesSelectMenu`.

File: frontend/src/v5/ui/controls/assigneesSelect/assigneesMultiSelect.component.tsx [hunk 1]
```diff
@@ -45,21 +45,20 @@ export const AssigneesSelect = ({
...
+}: IAssigneesMultiSelect) => {
 	const [open, setOpen] = useState(false);
 
 	// Must filter out users not included in this teamspace. This can occur when a user
 	// has been assigned to a ticket and later on is removed from the teamspace
 	const allUsersAndJobs = UsersHooksSelectors.selectUsersAndJobs();
-	const filteredValues = intersection(value, allUsersAndJobs.map((i) => i.user || i._id));
-
+	// eslint-disable-next-line max-len
+	const filteredValue: string[] = intersection(value, allUsersAndJobs.map((i) => i.user || i._id)) ?? [];
```
Comment (Amantini1997) [inline frontend/src/v5/ui/controls/assigneesSelect/assigneesMultiSelect.component.tsx:55]: `filteredValues` has been renamed and lost the `s` at the end, but it is an array, so I think the `s` good to indicate it was an collection

**Review by Amantini1997 — CHANGES_REQUESTED** [review]

**Comment by george-hadfield** [comment]
> This is a single select, but all the options have the checkbox, is this correct? image

You can have multiple users assigned though, so would need to visualise the previous selections

**Comment by george-hadfield** [comment]
The one of functionality I have not discussed with anyone. Where has this feature come from?

**Comment by The-Daniel** [comment]
@george-hadfield The oneOf functionality has been there from the start. This is just a custom oneOf (or manyOf) property where the values are 'jobsAndUsers' instead of an array of specified values. The same can be done for 'riskCategories'.

Before (when this issue's bug didn't happen) they looked like regular selects but I've added the new assignee select components. Partially out of ease since that component already exists now, and also because it condenses the information better than a default select. I should have run it by you first, let me know of any changes you'd like to be made

**Review by george-hadfield — APPROVED** [review]

**Review by sanmont3drepo — APPROVED** [review]

**Review by Amantini1997 — APPROVED** [review]

**Pull request merged**